src/connections/stream_api.rs

```diff
@@ -586,11 +587,15 @@ impl ConnectedStreamApi<state::Configured> {
 
         // Close worker threads
 
-        self.read_handle.await?;
-        self.write_handle.await?;
-        self.processing_handle.await?;
+        let (read_result, write_result, processing_result) =
+            join3(self.read_handle, self.write_handle, self.processing_handle).await;
 
-        trace!("TCP handlers fully disconnected");
+        // Note: we only return the first error.
+        read_result??;
+        write_result??;
+        processing_result??;
```

Comment:
Is the `??` something the user will need to do regularly? This feels like an annoying pattern

Reply:
This is an internal function and we can certainly make it nicer in a public interface, but I haven't finalized my thoughts on it.